oidc-interceptor.js

```diff
@@ -24,34 +24,40 @@ function getTokenState (token) {
   return TOKEN_STATE.VALID
 }
 
-function createOAuthInterceptor (options) {
-  const { refreshToken, clientId } = options
+function createOidcInterceptor (options) {
+  const { refreshToken, clientSecret, contentType } = options
   let {
-    accessToken ,
+    accessToken,
     retryOnStatusCodes,
-    origins
+    idpTokenUrl,
+    origins,
+    clientId,
   } = options
 
   retryOnStatusCodes = retryOnStatusCodes || [401]
   origins = origins || []
 
-  if (!refreshToken) {
-    throw new Error('refreshToken is required')
+  if (refreshToken) {
+    const decoded = decode(refreshToken)
+    const { iss, sub } = decoded
+
+    if (!iss) throw new Error('refreshToken is invalid: iss is required')
+
+    idpTokenUrl = idpTokenUrl || `${iss}/token`
+    clientId = clientId || sub
```

Comment:
I'm not sure what this is based on.

First of all `${iss}/token` cannot be a thing to rely upon.

Second, there is no mandate that a refresh_token be a JWT that can be decoded. It is an opaque artifact to everyone besides the Authorization Server.

You may have a conform endpoint discovery mechanism [like so](https://github.com/panva/oauth4webapi/blob/8173ba2944ede8beff11e59019940bbd6440ea96/src/index.ts#L1054-L1093) but the input to such would be the issuer identifier, not a token.

For all intents and purposes if a refresh token is passed in the other arguments must be the same. Plus refresh token rotation must be handled, the token endpoint call may return a new refresh token that replaces the one sent.

Reply:
Done, `idpTokenUrl` is now required. I've added a TODO for the discovery